pkg/sql/consistencychecker/consistency_checker.go

```diff
@@ -41,9 +41,7 @@ func (s *ConsistencyChecker) CheckConsistency(
 			EndKey: to,
 		},
 		Mode: mode,
-		// No meaningful diff can be created if we're checking the stats only,
-		// so request one only if a full check is run.
-		WithDiff: mode == roachpb.ChecksumMode_CHECK_FULL,
+		// NB: we're never requesting a diff since we couldn't render it anyway.
```

Comment:
Isn't the diff included in the reply? The `CheckConsistencyResponse_Result.Detail` field.

Reply:
Although, even if the `Detail` field is used in SQL, the possibility for it to contain the entire range worth of diff doesn't seam appealing.